aquadoggo/src/graphql/schema_builders/next_args.rs

```diff
@@ -1,56 +1,76 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 
-//! Static fields of the client api.
-use async_graphql::{Context, Object, Result};
-use p2panda_rs::api::next_args;
+use async_graphql::dynamic::{Field, FieldFuture, InputValue, Object, TypeRef};
+use dynamic_graphql::{FieldValue, ScalarValue};
+use log::debug;
+use p2panda_rs::api;
 use p2panda_rs::document::DocumentViewId;
 use p2panda_rs::identity::PublicKey;
 
 use crate::db::SqlStore;
-use crate::graphql::client::NextArguments;
-use crate::graphql::scalars;
-
-/// GraphQL queries for the Client API.
-#[derive(Default, Debug, Copy, Clone)]
-pub struct StaticQuery;
-
-#[Object]
-impl StaticQuery {
-    /// Return required arguments for publishing the next entry.
-    async fn next_args(
-        &self,
-        ctx: &Context<'_>,
-        #[graphql(
-            name = "publicKey",
-            desc = "Public key of author that will encode and sign the next entry \
-            using the returned arguments"
-        )]
-        public_key: scalars::PublicKeyScalar,
-        #[graphql(
-            name = "viewId",
-            desc = "Document the entry's UPDATE or DELETE operation is referring to, \
-            can be left empty when it is a CREATE operation"
-        )]
-        document_view_id: Option<scalars::DocumentViewIdScalar>,
-    ) -> Result<NextArguments> {
-        // Access the store from context.
-        let store = ctx.data::<SqlStore>()?;
-
-        // Convert and validate passed parameters.
-        let public_key: PublicKey = public_key.into();
-        let document_view_id = document_view_id.map(|val| DocumentViewId::from(&val));
-
-        // Calculate next entry's arguments.
-        let (backlink, skiplink, seq_num, log_id) =
-            next_args(store, &public_key, document_view_id.as_ref()).await?;
-
-        Ok(NextArguments {
-            log_id: log_id.into(),
-            seq_num: seq_num.into(),
-            backlink: backlink.map(|hash| hash.into()),
-            skiplink: skiplink.map(|hash| hash.into()),
+use crate::graphql::scalars::{DocumentViewIdScalar, PublicKeyScalar};
+use crate::graphql::types::NextArguments;
+
+const NEXT_ARGS_QUERY: &str = "nextArgs";
+const NEXT_ARGS_SCHEMA: &str = "NextArguments";
+const PUBLIC_KEY_ARG: &str = "publicKey";
+const PUBLIC_KEY_SCHEMA: &str = "PublicKey";
+const VIEW_ID_ARG: &str = "viewId";
+const VIEW_ID_SCHEMA: &str = "DocumentViewId";
```

Comment:
Similarily to the other comment it could be nice to explain what these constants are for in a docstring. Also, is the naming here differently or shouldn't it also be "_SCALAR" instead of "_SCHEMA"?